### ISSUE TITLE
Contour Plotting Improvements - [merged]

### ISSUE THREAD

**Issue by kratsg** [histfitter/histfitter#49]
In GitLab by @lawrenceleejr on Dec 4, 2018, 05:39

_Merges feature/contourPlottingImprovements -> master_

`harvestToContours.py`:

* Fixing a bug preventing debug contour plot from being drawn
* If debug on (`-d`), and upperLimit JSON present, output an upper limit surface for diagnostics

`multiplexContours.py`:

* Switched from local install of `shapely` to that provided in the SFT release `LCG_94` unless you have it installed already.

**Comment by kratsg** [comment]
In GitLab by @lawrenceleejr on Dec 4, 2018, 05:41

Tagging @pataraia, @jlorenz

**Comment by kratsg** [comment]
In GitLab by @jlorenz on Dec 4, 2018, 05:56

Hi @lawrenceleejr 
thanks for suggesting these changes and in particular also for removing the install procedure from inside the script. Would it be possible to add instructions about how to include/run the upper limits to the contour macro README as well (https://gitlab.cern.ch/HistFitter/HistFitter/blob/master/macros/Examples/contourPlotterExample/README.md), or is this not necessary?

Thanks,
Jeanette

**Comment by kratsg** [comment]
In GitLab by @lawrenceleejr on Dec 4, 2018, 06:02

added 1 commit

<ul><li>2d77abcb - contourPlottingExample: update to readme to include upper limit</li></ul>

[Compare with previous version](https://gitlab.cern.ch/HistFitter/HistFitter/merge_requests/35/diffs?diff_id=263256&start_sha=df940eab6c035c0b3ebd6ea0ca566fd69de09a4e)

**Comment by kratsg** [comment]
In GitLab by @lawrenceleejr on Dec 4, 2018, 06:03

For the record, the setup of `shapely` now in there was not possible until I requested that this package be included in the LCG release. Now that it is, we could switch to this kind of recommendation. I try to go with whatever is easiest for the user.

I've just pushed a new commit to update the readme with a small paragraph on the upper limit treatment.

-L
